pysrc/bytewax/execution.py

```diff
@@ -189,26 +459,199 @@ def run_cluster(
         worker_count_per_proc: Number of worker threads to start on
             each process.
 
-    Returns:
+        ipc_buffer_size: How big to make the IO queues when
+            communicating to subprocesses. Defautls to total number of
+            worker threads in cluster * 2.
+
+    Yields:
 
-        List of `(epoch, item)` tuples seen by capture operators.
+        `(epoch, item)` tuples seen by capture operators.
 
     """
-    man = Manager()
-    inp = man.list(list(inp))
+    worker_count = proc_count * worker_count_per_proc
+    if ipc_buffer_size is None:
+        ipc_buffer_size = worker_count * 2
+    manager = Manager()
+
+    main_raised_ex = manager.Event()
+    
+    in_q = manager.Queue(ipc_buffer_size)
+
+    def input_task():
+        logging.debug("Input task started")
+        for epoch_item in inp:
+            msg = ("yield", epoch_item)
+            msg_bytes = dill.dumps(msg)
+            in_q.put(msg_bytes)
```

Comment:
What would make this code much simpler, is an actual async IPC queue. Then we wouldn't need to spawn task threads and jump through hoops to unblock queues to cancel them.